tests/e2e/main_test.go

```diff
@@ -77,3 +84,110 @@ func TestLintMetrics(t *testing.T) {
 		t.Fatalf("the problems encountered in Lint are: %v", problems)
 	}
 }
+
+func TestDocumentation(t *testing.T) {
+	labelsDocumentation, err := getLabelsDocumentation()
+	if err != nil {
+		t.Fatal("Cannot get labels documentation", err)
+	}
+
+	buf := &bytes.Buffer{}
+
+	err = framework.KsmClient.metrics(buf)
+	if err != nil {
+		t.Fatal("Failed to get metrics from kube-state-metrics", err)
+	}
+
+	parser := &expfmt.TextParser{}
+	metricFamilies, err := parser.TextToMetricFamilies(buf)
+	if err != nil {
+		t.Fatal("Cannot decode metrics", err)
+	}
+
+	for _, metricFamily := range metricFamilies {
+		metric := metricFamily.GetName()
+
+		acceptedLabelNames, ok := labelsDocumentation[metric]
+		if !ok {
+			t.Errorf("Metric %v not found in documentation.", metric)
+			continue
+		}
+		for _, m := range metricFamily.Metric {
+			for _, l := range m.Label {
+				labelName := l.GetName()
+				labelNameMatched := false
+				for _, labelPattern := range acceptedLabelNames {
+					re, err := regexp.Compile(labelPattern)
+					if err != nil {
+						t.Errorf("Cannot compile pattern %v: %v", labelPattern, err)
+						continue
+					}
+					if re.MatchString(labelName) {
+						labelNameMatched = true
+						break
+					}
+				}
+				if !labelNameMatched {
+					t.Errorf("Label %v not found in documentation. Documented labels for metric %v are: %v",
+						labelName, metric, strings.Join(acceptedLabelNames, ", "))
+				}
+			}
+		}
+	}
+}
+
+// getLabelsDocumentation is a helper function that gets metric mabels documentation.
+// It returns a map where keys are metric names, and values are slices of label names,
+// and an error in case of failure.
+// By convention, UPPER_CASE parts in label names denotes wilcard patterns, used for dynamic labels.
+func getLabelsDocumentation() (map[string][]string, error) {
+	documentedMetrics := map[string][]string{}
+
+	docPath := "../../docs/"
+	docFiles, err := ioutil.ReadDir(docPath)
+	if err != nil {
+		return documentedMetrics, fmt.Errorf("failed to read documentation directory: %w", err)
+	}
+
+	// Match file names such as daemonset-metrics.md
+	fileRe := regexp.MustCompile(`^([a-z]*)-metrics.md$`)
+	// Match doc lines such as | kube_node_created | Gauge | `node`=&lt;node-address&gt;| STABLE |
+	lineRe := regexp.MustCompile(`^\| *(kube_[a-z_]+) *\| *[a-zA-Z]+ *\|(.*)\| *[A-Z]+`)
+	// Match label names in label documentation
+	labelsRe := regexp.MustCompile("`([a-zA-Z_][a-zA-Z0-9_]*)`")
+	// Match wildcard patterns for dynamic labels such as label_CRONJOB_LABEL
+	patternRe := regexp.MustCompile(`_[A-Z_]+`)
+
+	for _, file := range docFiles {
+		if file.IsDir() || !fileRe.MatchString(file.Name()) {
+			continue
+		}
+
+		filePath := path.Join(docPath, file.Name())
+		f, err := os.Open(filePath)
+		if err != nil {
+			return documentedMetrics, fmt.Errorf("cannot read file %v: %w", filePath, err)
```

Comment:
`%s` works for string slice too.
```suggestion
			return documentedMetrics, errors.Wrap("cannot read file %s", filePath)
```

Reply:
Ok, thanks